packages/account/src/Sections/Verification/ProofOfIncome/proof-of-income-form.tsx

```diff
@@ -42,7 +42,30 @@ const ProofOfIncomeForm = observer(({ onSubmit }: TProofOfIncomeForm) => {
     const { addNotificationMessageByKey, removeNotificationMessage, removeNotificationByKey } = notifications;
     const { isMobile, isDesktop } = useDevice();
 
-    const { upload } = useFileUploader();
+    const { upload, data, documentUploadStatus } = useDocumentUpload();
+
+    React.useEffect(() => {
+        const fetchAccountStatus = async () => {
+            if (documentUploadStatus === 'success') {
+                const get_account_status_response: DeepRequired<AccountStatusResponse> =
+                    await WS.authorized.getAccountStatus();
+                const { income, needs_verification } =
+                    get_account_status_response?.get_account_status?.authentication || {};
+                const needs_poinc =
+                    needs_verification.includes('income') &&
+                    [income_status_codes.REJECTED, income_status_codes.NONE].some(status => status === income?.status);
+                removeNotificationMessage({ key: 'needs_poinc' });
+                removeNotificationByKey({ key: 'needs_poinc' });
+                removeNotificationMessage({ key: 'poinc_upload_limited' });
+                removeNotificationByKey({ key: 'poinc_upload_limited' });
+                onSubmit(income?.status);
+                if (needs_poinc) {
```

Comment:
shouldnt we fetch latest getAccountsStatus before setting the notification?

Reply:
Once we get `success` from api response, we again make `get_account_status` call and store it in `get_account_status_response` (line 50)